rpc_jobs/release.yml

```diff
@@ -0,0 +1,120 @@
+- job:
+    name: RE-Release
+    project-type: pipeline
+    pipeline-scm:
+      scm:
+        - git:
+            url: https://github.com/rcbops/rpc-gating
+            branches:
+              - "${RPC_GATING_BRANCH}"
+            credentials-id: "github_account_rpc_jenkins_svc"
+      script-path: job_dsl/release.groovy
+    concurrent: false
+    properties:
+      - build-discarder:
+          num-to-keep: 30
+    parameters:
+      - rpc_gating_params
+      - text:
+          name: COMMAND
+          default: |
+            # the following parameters must be replaced with your values:
+            # REPO - the repo URL for the project to be released
+            # MAINLINE - the mainline branch, usually master
+            # REF - the rc/stabilisaiton branch, usually master-rc.
+            # VERSION - Symbollic name of the release, not the ref to be released
+            # SCRIPT - a script to execute that generates release notes
+            # PREVIOUS_VERSION - Used as base version for generating release notes
+            # NOTIFICATION_ADDRESS - where to send email release notifications
+            # Note: NOTIFICATION_ADDRESS must be in authorised recipients
+            #       list for the mailgun account, if using mailgun.
+            python rpc-gating/scripts/release.py \
+                --debug \
+                --org rcbops \
+                --repo REPO \
+                clone \
+                  --ref RC-BRANCH \
+                generate_release_notes \
+                    --script "gating/generate_release_notes/pre" \
```

Comment:
That confluence page is the definitive documentation at this point. Those three scripts were added to the job to comply with the standard set out on that page.

Reply:
`scripts/release.py` does not enforce the naming convention, it assumes it, but gives a specific error message when a script called `run` is used. Either we should not allow arbitrarily named scripts to run or error messages should be generalised.

Reply:
I'll add a --required-script parameter so that release.py doesn't have to know which scripts are required.

Reply:
Okay, but then I don't see how `release.py` can continue to have tests such as `if script.endswith("/run"):`.

Reply:
Specifying whether a script is required in parameters would be instead of release.py testing the script path.  I had a look at implementing --required-script, but that would make ordering difficult. 

```
generate_release_notes
  --script pre
  --required-script run
  --script post
```

The scripts and required-scripts would get collected into separate arrays, and the ordering would be lost.

Will try an `optional:` modifier within the script path. Eg `--script optional:/opt/myoptionalscript --script /opt/myrequiredscript` This ordering can be maintained while still communicating if the script is required or not.